### PR TITLE
[GCI-Ribhav Sharma] Connect to Database.

### PR DESCRIPTION
**Pull request by ribhavsharma**
## Task that I worked on
GCI Task: [GoLang-2: Connect to Database](https://codein.withgoogle.com/dashboard/task-instances/4755972690542592/)
## Description of what I changed
- [x]  Created an endpoint /patient/all which displays all patients in the db, in JSON format. Named the function GetAllPatients(c *gin.Context)
- [x]  Created endpoint /location/all to display all locations in JSON format. Named the function GetAllLocations(c *gin.Context)
- [x] Create endpoint /hospital/all to display all hospitals in JSON format. Name the function GetAllHospitals(c *gin.Context)
.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+	BirthDate   string `db:"birth_date json:"birth_date"`
+	LocationId  int    `db:"location_id" json:"location_id"`
+}
+
+/*
+connect with the database instance
+*/
+func initDatabase() {
+	var err error
+	db, err = sqlx.Open("postgres", "postgres://tueisrkv:4caHkg9pKgWatez67bwcydlM1C_U3o4B@rosie.db.elephantsql.com:5432/tueisrkv")
```
Comment (f4ww4z) [inline main.go]: Use `sqlx#Connect` or `sqlx#MustConnect` shorthand instead, so no need to ping.

File: main.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+		panic(err)
+	}
+
+}
+
+/*
+fetch location data from the database
+*/
+func GetAllLocations(c *gin.Context) {
+	location, error := db.Query("SELECT * from location")
```
Comment (f4ww4z) [inline main.go]: Use `db.Select()` to shorten your code and make it easier to read.

File: main.go [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+	if error != nil {
+		panic(error)
+	}
+
+	if error != nil {
```
Comment (f4ww4z) [inline main.go]: Duplicated if here.

File: main.go [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+	if error := location.Err(); error != nil {
+		panic(error)
+	}
+}
+
+/*
+fetch hospital data from the database
+*/
+func GetAllHospitals(c *gin.Context) {
+	hospital, error := db.Query("SELECT * from hospital")
```
Comment (f4ww4z) [inline main.go]: Use `db.Select()` to shorten your code and make it easier to read.

File: main.go [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+	if error != nil {
+		panic(error)
+	}
+
+	if error != nil {
```
Comment (f4ww4z) [inline main.go]: Duplicated `if` here.

File: main.go [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+	if error := hospital.Err(); error != nil {
+		panic(error)
+	}
+}
+
+/*
+fetch patient data from the database
+*/
+func GetAllPatients(c *gin.Context) {
+	patient, error := db.Query("SELECT * from patient")
```
Comment (f4ww4z) [inline main.go]: Use `db.Select()` to shorten your code and make it easier to read.

File: main.go [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+	if error != nil {
+		panic(error)
+	}
+
+	if error != nil {
```
Comment (f4ww4z) [inline main.go]: Duplicated `if` here

File: main.go (hunk 1 above)
Comment (ribhavsharma) [reply main.go]: done :)

File: main.go (hunk 3 above)
Comment (ribhavsharma) [reply main.go]: used ```db.Select()```, so this is fixed

File: main.go (hunk 4 above)
Comment (ribhavsharma) [reply main.go]: thanks for the suggestion!

File: main.go (hunk 6 above)
Comment (ribhavsharma) [reply main.go]: fixed

File: main.go (hunk 5 above)
Comment (ribhavsharma) [reply main.go]: removed

File: main.go (hunk 7 above)
Comment (ribhavsharma) [reply main.go]: fixed.

File: main.go (hunk 2 above)
Comment (ribhavsharma) [reply main.go]: done!

**Pull request merged**